### ISSUE TITLE
[type-annotation-spacing] Space between ? and :

### ISSUE THREAD

**Issue by Knutas** [typescript-eslint/typescript-eslint#3140]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/type-annotation-spacing": ["error"]
  }
}
```

```TS
function foo(a ? : string) {}
```

**Expected Result**

Expected error `Unexpected space before the ':'`.

**Actual Result**

No error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.6.0` |
| `@typescript-eslint/parser`        | `4.6.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.12.1` |
| `node`                             | `14.8.0` |
